web/src/lib/components/shared-components/base-modal.svelte

```diff
@@ -42,8 +40,8 @@
 >
   <div
     use:clickOutside
-    on:outclick={() => !ignoreClickOutside && dispatch('close')}
-    on:escape={() => dispatch('escape')}
+    on:outclick={() => !ignoreClickOutside && onClose && onClose()}
+    on:escape={() => onEscape && onEscape()}
```

Comment:
```suggestion
    on:escape={() => onEscape?.()}
```

Reply:
FYI: You can use the `?.` null-ish chaining operator even when invoking functions